cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/TranslationConfiguration.kt

```diff
@@ -228,6 +228,19 @@ private constructor(
         private var compilationDatabase: CompilationDatabase? = null
         private var matchCommentsToNodes = false
         private var addIncludesToGraph = true
+        private var passList =
```

Comment:
This list is not complete. I.e., there are language frontend-dependent passes (e.g. `JavaExternalTypeHierarchyResolver`, the `CompressLLVMPass`) or maybe a user might add another custom pass. If we could auto-generate this list/mapping somehow, this would be more cool :)

Reply:
All I can think of is some form of reflection that shows all Classes which inherit from Pass, also we then need a command line option that prints the list of available passes. When passes are then chosen they should be instantiable by getting the class from the provided class name. Required passes will have to be added as well. the question is if we print that information in the previously mentioned option that shows all available passes. @shenniger does that sound like something solving what you had in mind?

Reply:
Maybe we can explicitly state that these are only the "core" passes or something like that? I also see an issue by making this configurable because some passes *really* need to be run, otherwise things are just plain wrong (e.g. the language-frontend dependant passes). The only pass that really is sort of "optional" is the control-flow-sensitive DFG. And there we discussed about merging this with the regular DFG pass  and making the control flow sensitive stuff configurable.

TL;DR: I do not really see a use case for making the pass list configurable.

Reply:
I could see it for the use case where people implement their own passes, or in the future when our pass system is more diverse. However as you said, the dependent passes would have to be ran the same.

Reply:
@konradweiss, regarding auto-generation: I was looking at that, but it seems that there is no easy way to get the list of subclasses of a (non-sealed) Kotlin class. Hence the idea to store a list somewhere and then at least use reflection to get the names of the classes and instantiate. But maybe TranslationConfiguration is not a good place for such a list – we could put it somewhere more central, or allow for sublists per module – let me know what you think.

@oxisto regarding pass dependencies: The idea is to use this mostly for debugging and in cases where no full DFG is needed. But I agree that it should be in the docs that pass dependencies are still resolved with this, so you will likely get more passes than you asked for.